src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Entry<K, V>[] table;
+    private int size;
+    private int threshold;
+    private final float loadFactor;
+
+    static class Entry<K, V> {
+        final K key;
+        V value;
+        Entry<K, V> next;
+
+        public Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this(DEFAULT_INITIAL_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    public MyHashMap(int initialCapacity, float loadFactor) {
+        this.loadFactor = loadFactor;
+        this.threshold = (int) (initialCapacity * loadFactor);
+        this.table = new Entry[initialCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
 
-    }
+        for (Entry<K, V> entry = table[index]; entry != null; entry = entry.next) {
+            if ((entry.key == null && key == null) || (entry.key != null && entry.key.equals(key))) {
+                entry.value = value;
+                return;
+            }
+        }
+        addEntry(hash, key, value, index);
+        }
 
     @Override
-    public V getValue(K key) {
+    public V getValue (K key){
```

Comment:
Using public access modifiers everywhere is not a good practice. If the method 'getValue' is used only inside the same class, it should not be public. Consider changing it to private if that's the case.